js/README.md

```diff
@@ -43,3 +43,183 @@ There are several options that can be passed the function in the optional
   detecting an `@types` typings package is available.
+
+## Exploring the Entire Dependency Graph
+
+Imagine we have a project like this:
+
+```bash
+tree
+.
+├── components
+│   ├── first.ts
+│   └── second.ts
+├── deno.json
+└── someFiles
+    ├── c.ts
+    └── d.ts
+```
+
+with the following contents:
+
+- <details><summary>c.ts</summary>
+
+  ```ts
+  import { foo } from "@/someFiles/d.ts";
+
+  console.log(foo);
+  ```
+  </details>
+- <details><summary>d.ts</summary>
+
+  ```ts
+  import { val1 } from "@/components/first.ts";
+
+  console.log("from d.ts");
+  console.log({ val1 });
+
+  export const foo = "foo";
+  ```
+  </details>
+- <details><summary>first.ts</summary>
+
+  ```ts
+  import { val2 } from "./second.ts";
+
+  console.log("from first.ts");
+  console.log({ val2 });
+
+  export const val1 = 2;
+  ```
+  </details>
+- <details><summary>second.ts</summary>
+
+  ```ts
+  export const val2 = 2;
+  ```
+  </details>
+- <details><summary>deno.json</summary>
+
+  ```json
+  {
+    "imports": {
+      "@/": "./"
+    }
+  }
+  ```
+  </details>
+
+You can visualize the dependency graph like this:
+
+```mermaid
+graph LR
+    subgraph someFiles
+    c[c.ts] --> d[d.ts]
+    end
+
+    subgraph components
+    first[first.ts] --> second[second.ts]
+    end
+
+    d --> first
+```
+
+Let's see if we can see all these dependencies with the following snippet:
+
+```ts
+import { createGraph } from "https://deno.land/x/deno_graph@0.63.6/mod.ts";
+import { resolve, toFileUrl } from "https://deno.land/std@0.212.0/path/mod.ts";
+const path = resolve("./someFiles/c.ts");
+const graph = await createGraph(toFileUrl(path).href);
+console.log(graph);
+```
+
+We get the following output:
+
+```js
+{
+  roots: [
+    "file:///my/absolute/path/someFiles/c.ts"
+  ],
+  modules: [
+    {
+      kind: "esm",
+      dependencies: [ { specifier: "@/someFiles/d.ts", code: [Object] } ],
+      size: 58,
+      mediaType: "TypeScript",
+      specifier: "file:///my/absolute/path/someFiles/c.ts"
+    }
+  ],
+  redirects: {}
+}
+```
+
+Hmmm, that didn't go as well as hoped. It looks like we didn't traverse any of
+the transitive dependencies! Let's try again, but this time using the fantastic
+[`import_map`](https://github.com/denoland/import_map) project, which is used by
+the Deno CLI to resolve import maps.
+
+We'll load our import map from `deno.json` and then resolve it, using the root
+of our project (in this case, wherever the `deno.json` file is located). This
+library provides a `resolve` function which we can pass to `createGraph`.
+
+```ts
+import { createGraph } from "https://deno.land/x/deno_graph@0.63.6/mod.ts";
+import { resolve, toFileUrl } from "https://deno.land/std@0.212.0/path/mod.ts";
+import { parseFromJson } from "https://deno.land/x/import_map@v0.18.3/mod.ts";
+
+const path = resolve("./someFiles/c.ts");
+const importMap =
+  (await import("./deno.json", { with: { type: "json" } })).default;
```

Comment:
This won't work for jsonc deno.json files. Maybe this should use `Deno.readTextFileSync(new URL("./deno.json", import.meta.url))` instead?